src/librustc/mir/interpret/value.rs

```diff
@@ -237,13 +237,18 @@ impl<'tcx, Tag> Scalar<Tag> {
     }
 
     #[inline]
-    pub fn from_uint(i: impl Into<u128>, size: Size) -> Self {
+    pub fn try_from_uint(i: impl Into<u128>, size: Size) -> InterpResult<'tcx, Self> {
         let i = i.into();
-        assert_eq!(
-            truncate(i, size), i,
-            "Unsigned value {:#x} does not fit in {} bits", i, size.bits()
-        );
-        Scalar::Raw { data: i, size: size.bytes() as u8 }
+        if truncate(i, size) == i {
+            Ok(Scalar::Raw { data: i, size: size.bytes() as u8 })
+        } else {
+            throw_unsup_format!("Unsigned value {:#x} does not fit in {} bits", i, size.bits())
```

Comment:
I think this should return `Option<Self>`, and leave it up to the caller which error to raise (if any).